### PR TITLE
[FLINK-27643] Add new features to lifecycle management docs

### PR DESCRIPTION
**Pull request by gyfora**

### CODE REVIEW

**Review by morhidi — COMMENTED** [review]
+1 Nice docs, @gyfora. Love to see these cool new features written. nit: What are the followup steps in a manual recovery scenario once the job is up and running? Can we add a line or two for this?

File: docs/content/docs/custom-resource/job-management.md [hunk 1]
```diff
@@ -65,33 +66,38 @@ kubectl delete flinkdeployment my-deployment
...
 
-*\*In general no update can be executed while a savepoint operation is in progress*
 
-The three different upgrade modes are intended to support different use-cases:
- - stateless: Stateless applications, prototyping
- - last-state: Suitable for most stateful production applications. Quick upgrades in any application state (even for failing jobs), does not require a healthy job. Requires Flink Kubernetes HA configuration (see example below).
- - savepoint: Suitable for forking, migrating applications. Requires a healthy running job as it requires a savepoint operation before shutdown.
+*\*When Kubernetes HA is enabled the `savepoint` upgrade mode may fall back to the `last-state` behaviour in cases where the job is in an unhealthy state.*
+
+The three  upgrade modes are intended to support different scenarios:
```
Comment (Aitozi) [inline docs/content/docs/custom-resource/job-management.md:94]: extra space

**Review by Aitozi — COMMENTED** [review]
Can we also add the usage of the `restartNonce` and the session job managements together with the application job?

**Comment by gyfora** [comment]
@Aitozi I updated the docs with the rerstart info + added notes that these all should apply to sessionjobs as well. 
Also included some warnings where it's not the case (last-state, rollback not supported etc.)

**Review by Aitozi — APPROVED** [review]
Thanks, looks good to me +1

**Pull request merged**